src/Params.ml

```diff
@@ -13,6 +13,32 @@ module Format = Format_
 open Migrate_ast
 open Fmt
 
+let wrap_parens c ~opn_hint:(oh_space, oh_other) ~cls_hint:(ch_sp, ch_sl) k
+    =
+  let break_before, break_after =
+    let brk hint = fits_breaks "" ~hint "" in
+    match c.Conf.indicate_multiline_delimiters with
+    | `Space -> (brk oh_space, brk ch_sp)
+    | `No -> (brk oh_other, noop)
+    | `Closing_on_separate_line -> (brk oh_other, brk ch_sl)
+  in
+  str "(" $ break_before $ k $ break_after $ str ")"
+
+let wrap_beginend ~opn_hint:(_, oh) ~cls_hint:(_, ch) k =
+  str "begin"
+  $ fits_breaks "" ~hint:oh ""
+  $ k
+  $ fits_breaks "" ~hint:ch ""
+  $ str "end"
+
+let wrap_parens_or_beginend_if c ~parens_bch ~exp_grouping_bch ~opn_hint
```

Comment:
Similar logic is already implemented in `Fmt_ast` but I did not find it worth it to factorize it. What do you think?

Reply:
I think it would be nice to factorize this with `wrap_fits_breaks_exp_begin_end`, `parens_or_begin_end`, ` wrap_fits_breaks_exp_if` and `wrap_exp_if` if possible for sure, it makes sense to me to define these functions in Params since they depend on a lot of options.

Reply:
Moved parens wrapping to `Params`. Did not manage to share code with `if-then-else`.